plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -522,17 +522,20 @@ def make_template(in_str):
             # sliced_file to reflect the mapping of the added character(s) back
             # to the raw SQL.
             templated_sql = templated_sql + "\n" * n_trailing_newlines
-            sliced_file.append(
-                TemplatedFileSlice(
-                    slice_type="literal",
-                    source_slice=slice(
-                        len(source_dbt_sql) - n_trailing_newlines, len(source_dbt_sql)
-                    ),
-                    templated_slice=slice(
-                        len(templated_sql) - n_trailing_newlines, len(templated_sql)
-                    ),
+            if sliced_file[-1].templated_slice.stop != len(templated_sql):
```

Comment:
The new sanity checks caught this issue; apparently this bug had been here for months.

Reply:
Will `sliced_file` always contain at least one item? If not this code could fail with index out of bounds error couldn’t it?

Reply:
I think it will, but I'll add a guard just to be safe. 👍